trie/encoding_test.go

```diff
@@ -75,17 +99,51 @@ func TestHexKeybytes(t *testing.T) {
 	}
 }
 
-func BenchmarkHexToCompact(b *testing.B) {
-	testBytes := []byte{0, 15, 1, 12, 11, 8, 16 /*term*/}
-	for i := 0; i < b.N; i++ {
-		hexToCompact(testBytes)
+func TestBinaryKeybytes(t *testing.T) {
```

Comment:
```suggestion
func TestBinaryKeybytesWithoutPadding(t *testing.T) {
```
Mirroring the proceeding test if it makes sense

Reply:
What is meant by `padding` here? The keys themselves will never have padding. That concept is only important for [de]serialization to/from storage since we'll always store tightly-packed complete bytes but our key may be a number of bits that is not a multiple of 8.